shells/pipes-shell-2/pec-port.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {MessagePort} from '../../build/runtime/message-channel.js';
+
+class PecPort extends MessagePort {
+  constructor(arcId, bus) {
+    super();
+    this.arcId = arcId;
```

Comment:
Messages that come in over the bus are already mapped to arcs. I suspect there is a simpler way to pull this off.